pkg/reconciler/common/read_releases.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+
+	"golang.org/x/mod/semver"
+)
+
+// RetrieveManifestPath returns the manifest path for Knative based a provided version and component
+func RetrieveManifestPath(ctx context.Context, version, kcomponent string) string {
+	koDataDir := os.Getenv("KO_DATA_PATH")
+	return filepath.Join(koDataDir, kcomponent, version)
```

Comment:
@jcrossley3 What do you mean by an unknown version? Is it a version that is not semantic version, or not available bundled in the image?

Reply:
I meant unknown as in "the operator doesn't know of a valid path to a bag of yaml representing that version". I was thinking the operator would just see if `os.Stat(result)` returned an error before returning the result, just because it's easy enough to verify that the version matches an actual embedded dir. I'm not sure I'd want that behavior for non-local path, e.g. http, so we can revisit later, or if you feel strongly that we shouldn't ensure the path is valid, that's fine, too.

Reply:
We should OK without validating the path.
When we create the manifest, it will surely returns an error, if the path we get from this function is not valid.

Reply:
Should probably have a test validating that

Reply:
tests added to validate the manifest creation.